src/pages/Login/Login.jsx

```diff
@@ -1,160 +1,151 @@
-    setstate({ ...state, Password: event.target.value });
-  };
-
-  const hasError = () => {
-    try {
-      return !schema.validateSync(state);
-    } catch (err) {
-      return true;
+    constructor(props) {
+      super(props);
+      this.state = {
+        Email: '',
+        Password: '',
+        touched: {
+          email: false,
+          password: false,
+        },
+      };
     }
-  };
-
-  const handleBlur = (field) => {
-    setblur({ ...blur, [field]: true });
-  };
-
-  const isTouched = () => (blur.Password || blur.Email);
 
-  const handleClose = () => {
-    setopen({ ...open, openDialog: false });
-    console.log(state);
-  };
+    handleChange = (key) => ({ target: { value } }) => {
+      this.setState({ [key]: value });
+    };
 
-  const getError = (field) => {
-    if (blur[field] && hasError()) {
+    hasErrors = () => {
       try {
-        schema.validateSyncAt(field, state);
+        this.schema.validateSync(this.state);
       } catch (err) {
-        return err.message;
+        return true;
       }
+      return false;
     }
-    return null;
-  };
 
-  return (
-    <Container className={classes.container} fullWidth aria-labelledby="form-dialog-title">
-      <LockRoundedIcon className={classes.icon} />
-      <Typography id="form-dialog-title" className={classes.login}>Login</Typography>
-      <TextField
-        className={classes.input}
-        size="medium"
-        id="Email"
-        label="Email"
-        type="email"
-        variant="outlined"
-        error={getError('Email')}
-        helperText={getError('Email')}
-        onChange={handleEmailChange}
-        onBlur={() => handleBlur('Email')}
-        fullWidth
-        InputProps={{
-          startAdornment: (
-            <InputAdornment position="start">
-              <MailIcon />
-            </InputAdornment>
-          ),
-        }}
-      />
-      <TextField
-        className={classes.input}
-        size="medium"
-        id="Password"
-        label="Password"
-        type="password"
-        variant="outlined"
-        error={getError('Password')}
-        helperText={getError('Password')}
-        onChange={handlePasswordChange}
-        onBlur={() => handleBlur('Password')}
-        InputProps={{
-          startAdornment: (
-            <InputAdornment position="start">
-              <VisibilityOffIcon />
-            </InputAdornment>
-          ),
-        }}
-        fullWidth
-      />
-      <Button fullWidth onClick={handleClose} className={classes.signin} color="primary" variant="contained" disabled={hasError() || !isTouched()}>
-        SIGN IN
-      </Button>
-    </Container>
-  );
-};
+    // eslint-disable-next-line consistent-return
+    getError = (field) => {
+      const { touched } = this.state;
+      if (touched[field] && this.hasErrors()) {
+        try {
+          this.schema.validateSyncAt(field, this.state);
+          return '';
+        } catch (err) {
+          return err.message;
```

Comment:
why validate schema twice, validate schema and create error-handler to handle errors